apps/src/templates/instructions/TopInstructionsCSF.jsx

```diff
@@ -176,40 +171,64 @@ var TopInstructions = React.createClass({
 
   getInitialState() {
     return {
-      rightColWidth: 90,
-      promptForHint: false
+      rightColWidth: this.shouldDisplayCollapserButton() ? 90 : 10,
+      promptForHint: false,
+      displayScrollButtons: true
     };
   },
 
-  componentDidUpdate(prevProps) {
+  componentDidUpdate(prevProps, prevState) {
     // Update right col width now that we know how much space it needs, and
     // rerender if it has changed. One thing to note is that if we end up
     // resizing our column significantly, it can result in our maxNeededHeight
     // being inaccurate. This isn't that big a deal except that it means when we
     // adjust maxNeededHeight below, it might not be as large as we want.
-    const width = $(ReactDOM.findDOMNode(this.refs.collapser)).outerWidth(true);
+    const width = this.shouldDisplayCollapserButton() ?
+        $(ReactDOM.findDOMNode(this.refs.collapser)).outerWidth(true) : 10;
     if (width !== this.state.rightColWidth) {
       // eslint-disable-next-line react/no-did-update-set-state
       this.setState({
         rightColWidth: width
       });
     }
 
-    const gotNewFeedback = this.props.feedback && !prevProps.feedback;
+    this.adjustMaxNeededHeight();
+
+    if (this.refs && this.refs.instructions) {
+      const contentContainer = this.refs.instructions.parentElement;
+      const contentHeight = contentContainer.scrollHeight;
+      const canScroll = contentContainer.scrollHeight > contentContainer.clientHeight;
+      if (canScroll !== this.state.displayScrollButtons) {
+        // eslint-disable-next-line react/no-did-update-set-state
+        this.setState({
+          displayScrollButtons: canScroll
+        });
+      }
+    }
+
+    const gotNewHint = prevProps.hints.length !== this.props.hints.length;
+    if (gotNewHint) {
+      const images = ReactDOM.findDOMNode(this.refs.instructions).getElementsByTagName('img');
+      for (const image of images) {
+        image.onload = image.onload || this.scrollInstructionsToBottom;
```

Comment:
If there's an existing handler, don't you want it to call both the original `onload` and then `scrollInstructionsToBottom`, rather than just one or the other?